### PR TITLE
feature: improved receipt page

### LINKED ISSUE #4595: feature: improve design of receipt page for Sequoia form template
# Feature Request

## User Story
As a user, I want the Sequoia form template to display a more informative and interesting receipt page so that I am given consistent experience all the way through to the end of my donation.

This receipt page should also include social sharing buttons than can be hidden or shown, dependent on a setting in form template options. 

## Visuals
See Figma boards: https://www.figma.com/file/jaVXDA4wjQD9OqTi0l9KQe/Form-Themes?node-id=1882%3A44&viewport=-259%2C1586%2C0.5

## Acceptance Criteria
- [ ] Receipt page design matches Figma design (see Visuals sections)
- [ ] Social sharing buttons are conditionally shown or hidden (depending on setting in template options)
- [ ] Default SVG image color matches form's primary color
- [ ] Page includes donation details as outlined on Figma

### PR DESCRIPTION
**Pull request by henryholtgeerts**
## Description
Resolves #4595 
This pull request introduces an improved receipt page to the Sequoia form template. Based on Figma designs, the receipt page template displays data relating to the recently processed payment, as well as relevant dynamic content pulled from template options.

The payment ID is retrieved using a new `getPaymentId` helper method. This ID is used to instantiate a new `Give_Payment` which becomes the source for payment information displayed.

Template options are loaded using the `getThemeOptions` method, and determine the rendering of certain markup. 

## Affects
This PR primarily impacts the Sequoia form template, introducing a new `receiptTemplate` file that handles receipt page markup. The PR also adds a new `getPaymentId` helper function to the Theme Frontend Utilities. It also introduces a number of classes to the Sequoia CSS assets specific to receipt page styling. Finally, the PR introduces the 'donor_email' email tag.

## What to test
Test out the various form template options found in the Thank You section. Make sure that they work as expected. See how changes in the form's primary color impact the Thank You page. Compare the design of the Thank You page to those in Figma (see Figma: https://www.figma.com/file/jaVXDA4wjQD9OqTi0l9KQe/Form-Themes?node-id=1882%3A44&viewport=-259%2C1586%2C0.5).

_**NOTE:** The image option for the thank you page section displays differently than indicated by the templat option description. This warrants further discussion, as from a UX perspective in may make more sense to allow users to use any image (as they would for the introduction image) than anticipate they would want to use some kind of checkmark or icon here. Presently, this PR includes the thank you image, in the same way it includes the introduction image. As it is currently, the checkmark displays in the form's primary color if no image is provided._

## Screenshots:
Screen Shot 2020-04-09 at 3 10 58 PM

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: src/Views/Form/Themes/Sequoia/Sequoia.php [hunk 1]
```diff
@@ -12,6 +12,18 @@
...
  */
 class Sequoia extends Theme implements Hookable, Scriptable {
+	
+	/**
+	 * Sequoia templates
+	 *
+	 * @since 2.7.0
+	 * @var array
+	 */
+	public $templates = [
```
[the remark this answers is not in the document]
Comment (JasonTheAdams) [reply src/Views/Form/Themes/Sequoia/Sequoia.php:22]: Let's hop on a call today and take a look a how this property works and is intended to be extended.

File: src/Helpers/Form/Theme/Utils/Frontend.php [hunk 2]
```diff
@@ -61,4 +61,8 @@ function getFormId() {
 	return null;
 }
 
+function getPaymentId() {
```
[the remark this answers is not in the document]
Comment (henryholtgeerts) [reply src/Helpers/Form/Theme/Utils/Frontend.php:76]: Done, ref: https://github.com/impress-org/givewp/pull/4615/commits/47b2c2342594a15071af1436aea1c61e6f8ccff2

File: src/Helpers/Form/Theme/Utils/Frontend.php [hunk 3]
```diff
@@ -61,4 +61,8 @@ function getFormId() {
...
+	$session = give_get_purchase_session();
+	return $session['donation_id'];
```
[the remark this answers is not in the document]
Comment (henryholtgeerts) [reply src/Helpers/Form/Theme/Utils/Frontend.php]: Done, ref: https://github.com/impress-org/givewp/pull/4615/commits/a22e955d178339ca946b2b9cf1b1f7a9131d9b29

File: src/Views/Form/Themes/Sequoia/Sequoia.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (henryholtgeerts) [reply src/Views/Form/Themes/Sequoia/Sequoia.php]: This has been address with the new view getter methods, ref: https://github.com/impress-org/givewp/pull/4615/commits/181c392dde77aad170ca50e00580d199c3e7a682

**Comment by henryholtgeerts** [comment]
> I want to be clear here that there is no pre-existing receipt view that we're reusing here, right? As opposed to other areas of the form. That's why we're building out the full view markup?

Yes, that's correct. Because the receipt page is loaded separate from the form view, it needs full view markup. This may be different after @ravinderk 's latest work on building a "skin generator" - but I will need to take a look at that.

**Comment by JasonTheAdams** [comment]
Got it, thank you for catching me up to speed. I believe Ravinder's PR for the scaffolding (I just can't call it skin) is merged. If that's the case, go ahead and merge `develop` into this branch and update to use that system.

**Comment by ravinderk** [comment]
@henryholtgeerts When you see legacy receipt template then you notice that receipt first shows a placeholder and then we render receipt after getting it by ajax request. This logic prevents the donor to show cached receipt accidentally. Can we reuse same concept here?

Let me know if you have a question.

cc: @DevinWalker

**Comment by DevinWalker** [comment]
@henryholtgeerts This looks good except one minor issue:



The margin should match the margin below ~30 pixels top and bottom. 

Can you adjust that? Then it's good to merge on my end.

**Comment by henryholtgeerts** [comment]
> @henryholtgeerts This looks good except one minor issue:
> 
> 
> 
> The margin should match the margin below ~30 pixels top and bottom.
> 
> Can you adjust that? Then it's good to merge on my end.

Taken care of, ref: https://github.com/impress-org/givewp/pull/4615/commits/8a5619b17b4a423f1ce99944fb459f6c42b3d0d6

**Comment by DevinWalker** [comment]
@henryholtgeerts I also noticed some extra padding where the text from the concept is supposed to be:

*Concept*



*Preview from this PR*



Also...

I noticed there's no UI to customize the text that appears above the social media sharing text. Can we add that if this option is enabled?

File: src/Views/Form/Templates/Sequoia/assets/css/form.scss [hunk 4]
```diff
@@ -517,6 +533,153 @@ p {
 	}
 }
 
+.receipt {
```
Comment (ravinderk) [inline src/Views/Form/Templates/Sequoia/assets/css/form.scss:536]: @henryholtgeerts Can we create a separate file for receipt styling?

Comment (henryholtgeerts) [reply src/Views/Form/Templates/Sequoia/assets/css/form.scss]: Done, ref: https://github.com/impress-org/givewp/pull/4615/commits/c907455f18f499073b462907f6663589c5319428

**Pull request closed without merging**